components/camel-jpa/src/main/java/org/apache/camel/component/jpa/JpaProducer.java

```diff
@@ -124,8 +124,10 @@ private Object save(final Object entity) {
                  */
                 private Object remove(final Object entity) {
                     LOG.debug("remove: {}", entity);
-                    entityManager.remove(entity);
-                    return entity;
+
+                    Object managedEntity = entityManager.contains(entity) ? entity : entityManager.merge(entity);
```

Comment:
This may confuse users reading this code why its doing this code. Can you add some code comments.
Also it seems JPA do not have a good way to re-attach an entity accordingly to SO users: http://stackoverflow.com/questions/912659/what-is-the-proper-way-to-re-attach-detached-objects-in-hibernate.